test/e2e/upgrades/configmaps.go

```diff
@@ -135,12 +155,33 @@ func (t *ConfigMapUpgradeTest) testPod(f *framework.Framework) {
 		},
 	}
 
-	expectedOutput := []string{
-		"content of file \"/etc/configmap-volume/data\": some configmap data",
-		"mode of file \"/etc/configmap-volume/data\": -rw-r--r--",
+	pod, err := f.ClientSet.Core().Pods(t.configMap.Namespace).Create(pod)
+	framework.ExpectNoError(err)
+
+	return pod, err
+}
+
+// validate that the ConfigMap data is read correctly using both methods
+func (t *ConfigMapUpgradeTest) testPod(f *framework.Framework) {
```

Comment:
This has fundamentally changed the test now, and I'm afraid it will no longer work during a cluster upgrade. It now relies on a pod you create in `Setup` instead of creating one itself. So during a cluster upgrade, the pod created in `Setup` will have its node upgraded and will no longer be running. Then this command will run after the upgrade and not be able to exec inside the pod.

This workflow is a little complicated to get right. I suggest either using a deployment to have it automatically rescheduled, or making sure a pod is running first, and if not, create it.

And since this is going to be reworked a little, could you do this change in a separate PR? I'd really like to get your daemon set test in. :smile: